Orange/preprocess/impute.py

```diff
@@ -26,7 +28,65 @@ def transform(self, c):
         return numpy.where(numpy.isnan(c), self.value, c)
 
 
-class Average:
+class BaseImputeMethod:
+    name = ""
+    short_name = ""
+    description = ""
+    format = "{var.name} -> {short_name}"
+    columns_only = False
+
+    @abstract
+    def __call__(self, data, variable):
+        """ Imputes table along variable column.
+
+        Args:
+            data (Table): A table to impute.
+            variable (Variable): Variable for completing missing values.
+
+        Returns:
+            A new Variable instance with completed missing values or
+            a array mask of rows to drop out.
+        """
+        raise NotImplementedError()
+
+    def format_variable(self, var):
+        return self.format.format(var=var, short_name=self.short_name)
+
+    def __str__(self):
+        return self.name
+
+    def copy(self):
+        return self
+
+    @classmethod
+    def check_supports_variable(cls, variable):
+        return True
+
+
+class DoNotImpute(BaseImputeMethod):
+    name = "Don't impute"
+    short_name = "leave"
+    description = ""
+
+    def __call__(self, data, variable):
+        return variable
+
+
+class DropInstances(BaseImputeMethod):
+    name = "Remove instances with unknown values"
+    short_name = "drop"
+    description = ""
+
+    def __call__(self, data, variable):
+        index = data.domain.index(variable)
+        return numpy.isnan(data[:, index]).reshape(-1)
+
+
+class Average(BaseImputeMethod):
+    name = "Average/Most frequent"
+    short_name = "average"
+    description = "Replace with average/modus for the column"
```

Comment:
Modus is [mode](https://en.wikipedia.org/wiki/Mode_%28statistics%29) in English.